api/src/main/java/org/pmiops/workbench/billing/InitialCreditsService.java

```diff
@@ -242,6 +258,155 @@ public double getWorkspaceCreatorFreeCreditsRemaining(DbWorkspace dbWorkspace) {
     return Math.max(creatorFreeCreditsRemaining, 0);
   }
 
+  public void checkCreditsExpirationForUserIDs(List<Long> userIdsList) {
+    if (userIdsList != null && !userIdsList.isEmpty()) {
+      Iterable<DbUser> users = userDao.findAllById(userIdsList);
+      users.forEach(this::checkExpiration);
+    }
+  }
+
+  public Optional<Timestamp> getCreditsExpiration(DbUser user) {
+    return Optional.ofNullable(user.getUserInitialCreditsExpiration())
+        .filter(exp -> !exp.isBypassed()) // If the expiration is bypassed, return empty.
+        .filter(exp -> !institutionService.shouldBypassForCreditsExpiration(user))
+        .map(DbUserInitialCreditsExpiration::getExpirationTime);
+  }
+
+  public boolean haveCreditsExpired(DbUser user) {
```

Comment:
took me a while to understand what does this  `have` mean.
(1) User has many credits, and one of them expired
(2) User credits already expired. 
Looks that is 2. Wondering if we can have a better name for it

Reply:
Thanks for pointing this out. How does `areUserCreditsExpired` sound?